tests/components/prometheus/test_init.py

```diff
@@ -33,24 +38,109 @@ class FilterTest:
     should_pass: bool
 
 
-async def prometheus_client(hass, hass_client, namespace):
+async def setup_prometheus_client(hass, hass_client, namespace):
     """Initialize an hass_client with Prometheus component."""
+    # Reset registry
+    prometheus_client.REGISTRY = prometheus_client.CollectorRegistry(auto_describe=True)
+    prometheus_client.ProcessCollector(registry=prometheus_client.REGISTRY)
+    prometheus_client.PlatformCollector(registry=prometheus_client.REGISTRY)
+    prometheus_client.GCCollector(registry=prometheus_client.REGISTRY)
+
     config = {}
     if namespace is not None:
         config[prometheus.CONF_PROM_NAMESPACE] = namespace
-    await async_setup_component(hass, prometheus.DOMAIN, {prometheus.DOMAIN: config})
+    assert await async_setup_component(
+        hass, prometheus.DOMAIN, {prometheus.DOMAIN: config}
+    )
+    await hass.async_block_till_done()
 
-    await async_setup_component(hass, sensor.DOMAIN, {"sensor": [{"platform": "demo"}]})
+    return await hass_client()
 
-    await async_setup_component(
-        hass, climate.DOMAIN, {"climate": [{"platform": "demo"}]}
+
+async def generate_latest_metrics(client):
+    """Generate the latest metrics and transform the body."""
+    resp = await client.get(prometheus.API_ENDPOINT)
+    assert resp.status == HTTPStatus.OK
+    assert resp.headers["content-type"] == CONTENT_TYPE_TEXT_PLAIN
+    body = await resp.text()
+    body = body.split("\n")
+
+    assert len(body) > 3
+
+    return body
+
+
+async def test_view_empty_namespace(hass, hass_client):
+    """Test prometheus metrics view."""
+    client = await setup_prometheus_client(hass, hass_client, "")
+
+    sensor2 = DemoSensor(
```

Comment:
There's no reason to create real entities for these tests as prometheus just listens for state change events. Setting a state in the state machine is enough. Would be good to rework this in the future.